src/main/java/com/mraof/minestuck/item/crafting/alchemy/generator/DefaultInterpreter.java

```diff
@@ -33,9 +37,37 @@ public GristSet generateCost(IRecipe<?> recipe, Item output, GenerationContext c
 	{
 		if(recipe.isSpecial())
 			return null;
-		
+
 		GristSet totalCost = new GristSet();
-		for(Ingredient ingredient : recipe.getIngredients())
+
+		// Frick reflection frick reflection frick reflection frick reflection i hate java
+		// this was a whole minecraft oversight anyway
+		// so if they eventually fix this, remove this bit ok?
+		NonNullList<Ingredient> ingredients;
+		if (recipe instanceof SmithingRecipe)
```

Comment:
Can you write this as a new recipe interpreter for smithing recipes instead of extending the default interpreter?